bibliofil_lists.module

```diff
@@ -0,0 +1,293 @@
+<?php
+
+/**
+ * @file
+ * Bibliofil lists module.
+ */
+
+define('DYNPATH', '/cgi-bin/m2dyn.htmc');
+define('WEBLISTERPATH', '/cgi-bin/rest_service/web_lister/1.0/data/');
+
+/**
+ * Implements hook_ctools_plugin_directory().
+ */
+function bibliofil_lists_ctools_plugin_directory($owner, $plugin_type) {
+  if ($owner == 'ctools' || $owner == 'panels') {
+    return 'plugins/' . $plugin_type;
+  }
+}
+
+/**
+ * Implements hook_menu().
+ */
+function bibliofil_lists_menu() {
+  $items = [];
+
+  $items['admin/config/bibliofil'] = [
+    'title' => 'Bibliofil',
+    'description' => 'Bibliofil specific settings',
+    'page callback' => 'system_admin_menu_block_page',
+    'access arguments' => ['access administration pages'],
+    'file' => 'system.admin.inc',
+    'file path' => drupal_get_path('module', 'system'),
+  ];
+
+  $items['admin/config/bibliofil/bibliofil_lists'] = [
+    'title' => 'Bibliofil lists',
+    'description' => 'Bibliofil lists specific settings',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => ['bibliofil_lists_admin_settings_form'],
+    'file' => 'bibliofil_lists.admin.inc',
+    'access arguments' => ['access administration pages'],
+  ];
+
+  $items['bibliofil/%/%'] = [
+    'type' => MENU_CALLBACK,
+    'page callback' => 'bibliofil_lists_ajax_callback',
+    'page arguments' => [1, 2],
+    'access arguments' => ['access content'],
+    'theme callback' => 'ajax_base_page_theme',
+    'delivery callback' => 'ajax_deliver',
+  ];
+
+  return $items;
+}
+
+/**
+ * @param $hash
+ * @param $type
+ *
+ * @throws \Exception
+ */
+function bibliofil_lists_ajax_callback($hash, $type) {
+  if (empty($hash) || empty($type)) {
+    return;
+  }
+
+  if ($type == 'librarylists') {
+    $element = bibliofil_lists_build_librarylists($hash);
+  }
+
+  if ($type == 'suggestions') {
+    $element = bibliofil_lists_build_suggestions($hash);
+  }
+
+  echo render($element);
+  drupal_exit();
+}
+
+/**
+ * Request and build suggestions carousel.
+ *
+ * @param string $hash
+ *   Request hash.
+ *
+ * @return array|null
+ *   Renderable array or empty result.
+ *
+ * @throws \Exception
+ */
+function bibliofil_lists_build_suggestions($hash) {
+  $opac_url = variable_get('bibliofil_lists_opac_url', '');
+
+  $decoded = base64_decode($hash);
+  $explode = explode(':', $decoded);
+
+  $conf_toplister = $explode[0];
+  $conf_ants = $explode[1];
+  $conf_ante = $explode[2];
+  $conf_instans = $explode[3];
+  $conf_list_type = $explode[4];
```

Comment:
There's no proof that these might exist. Please check the existence of those keys, by simply:
`$conf_toplister = $explode[0] ?? null;` or `$conf_toplister = $explode[0] ?? '';`